src/lib.rs

```diff
@@ -158,18 +182,20 @@ where
 /// let mut private_key = vec![0u8; 32];
 /// rng.fill_bytes(private_key.as_mut_slice());
 ///
-/// let uuid = encrypt_key(&dir, &mut rng, &private_key, "password_to_keystore")?;
+/// let uuid = encrypt_key_with_id(&dir, Uuid::new_v4().to_string(), &mut rng, &private_key, "password_to_keystore")?;
 /// # Ok(())
 /// # }
 /// ```
-pub fn encrypt_key<P, R, B, S>(
+pub fn encrypt_key_with_id<P, I, R, B, S>(
     dir: P,
+    id: I,
```

Comment:
I actually don't mind adding a breaking change, I will publish a new minor version.

```suggestion
    id: Option<I>,
```

Reply:
I would prefer the `id` being the last argument as well, just because its the only optional argument.